pkg/resource/sensitive.go

```diff
@@ -124,23 +126,37 @@ func GetSensitiveAttributes(from map[string]interface{}, mapping map[string]stri
 			if v == nil {
 				continue
 			}
-			s, ok := v.(string)
-			if !ok {
+
+			switch s := v.(type) {
+			case []interface{}:
+				for i, e := range s {
+					// Note(turkenh): k8s secrets uses a strict regex to validate secret
+					// keys which does not allow having brackets inside. So, we need to
+					// do a conversion to be able to store as connection secret keys.
+					// See https://github.com/crossplane/terrajet/pull/94 for
+					// more details.
+					k, err := fieldPathToSecretKey(fp)
+					if err != nil {
+						return nil, errors.Wrapf(err, "cannot convert fieldpath %q to secret key", fp)
+					}
+					k = strings.TrimSuffix(k, pluralSuffix)
+					if vals == nil {
+						vals = map[string][]byte{}
+					}
+					vals[fmt.Sprintf("%s%s.%d", prefixAttribute, k, i)] = []byte(fmt.Sprintf("%v", e))
```

Comment:
Is there a specific reason to use `fmt.Sprintf("%v", e)` instead of a [type assertion](https://go.dev/ref/spec#Type_assertions) to string? 

I don't have a strong opinion but a little concerned because this would not be consistent with how we are handling single element, e.g. [fail](https://github.com/crossplane/terrajet/pull/197/files?diff=unified&w=0#diff-aee63bee02430ee5f3effa8c4f524ebc94fea5e2b200b5b4b53187ed1bc440fdR158) if not a string.